src/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacet.ts

```diff
@@ -254,7 +254,27 @@ export class DynamicHierarchicalFacet extends Component implements IDynamicHiera
      *
      * @examples electronics, electronics\,laptops
      */
-    basePath: ComponentOptions.buildListOption<string>({ defaultValue: [] }),
+    basePath: ComponentOptions.buildListOption<string>({ defaultValue: [], section: 'CommonOptions' }),
+
+    /**
+     * Whether to use the [`basePath`]{@link DynamicHierarchicalFacet.options.basePath} option as a filter for the results.
+     *
+     * **Note:**
+     * Setting this option to `true` and using the [`dependsOn`]{@link DynamicHierarchicalFacet.options.dependsOn} option can cause unexpected behaviour.
+     */
+    basePathShouldFilterResults: ComponentOptions.buildBooleanOption({
+      defaultValue: false,
+      depend: 'basePath',
+      section: 'CommonOptions',
+      postProcessing: (value, options: IDynamicHierarchicalFacetOptions) => {
+        if (value && options.dependsOn) {
+          new Logger(options).warn('basePathShouldFilterResults is not compatible with dependsOn, set to false');
```

Comment:
Hmm, I'm not clear on what the incompatibility is between the `basePath` and `dependsOn` options?